include/spdlog/logger.h

```diff
@@ -84,10 +84,15 @@ class SPDLOG_API logger
     logger &operator=(logger other) SPDLOG_NOEXCEPT;
     void swap(spdlog::logger &other) SPDLOG_NOEXCEPT;
 
+    void log(source_loc loc, level::level_enum lvl, std::initializer_list<Field> fields, string_view_t msg)
```

Comment:
I am correct in that this will only support a flat json object?  No support for nested json objects or json arrays?

Reply:
Yes, that is correct.  It's high in the list of TODOs in the README.

I toyed around with different interfaces.  I've considered writing some glue for objects that are serializable to https://github.com/nlohmann/json (or other popular libraries), but I hate to add dependencies for spdlog (I'm also not sure how performant the interfaces are).

Data is currently stuffed into a big ol' union of fundamental C++ types; if we suppotred lists and objects, there would have to be additional allocations for data (at least for multithreaded loggers).

@cfxegbert  - what kind of interfaces would you like to see?

Reply:
I am currently using https://github.com/takeshibaconsuzuki/structlog in my current project which uses nlohmann::json.  I like your approach of having it as an argument to the log call instead of returning a context object I modified.

 I have a visitor on a std::variant that returns a json object.
```
    m_logger->debug("command response")({
        {"command", enumToString(Command(response.command))},
        {"errorCode", enumToString(response.errorCode)},
        {"result", std::visit(VisitResult(), response.result)}
    });
```
```
struct VisitResult {
    auto operator()(const bool val) const noexcept {
        return nlohmann::json(val);
    }
    auto operator()(const std::uint16_t val) const noexcept {
        return nlohmann::json(toHex(val));
    }

    ...

    auto operator()(const Murideo::Timing &timing) const noexcept {
        return nlohmann::json({{"Timing", enumToString(timing)}});
    }
    auto operator()(const Murideo::RGBPatternStatus &pattern) const noexcept {
        return nlohmann::json({
            {"color", {
                          float(pattern.red)/255.0F,
                          float(pattern.green)/255.0F,
                          float(pattern.blue)/255.0F
                      }},
            {"background", {
                               float(pattern.redBackground)/255.0F,
                               float(pattern.greenBackground)/255.0F,
                               float(pattern.blueBackground)/255.0F
                           }},
            {"coverage", pattern.percent},
            {"hdr", pattern.hdr}
        });
    }

    ...

}
```
As you can see I am returning just a json value or a nested object from my visitor.  The fields `color` and `background` on the pattern status are lists of floats.